library/src/compute/calc.rs

```diff
@@ -94,13 +94,18 @@ pub fn pow(
         Num::Int(i) if i > u32::MAX as i64 => {
             bail!(exponent.span, "exponent too large");
         }
-        Num::Int(0..) => exponent.v,
-        Num::Float(f) if f >= 0.0 => exponent.v,
+        Num::Int(i) if i != 0 => exponent.v,
+        Num::Float(f) if f.is_normal() => exponent.v,
         _ => {
-            bail!(exponent.span, "exponent must be non-negative");
+            bail!(exponent.span, "exponent must be normal (non-zero, non-NaN, non-infinite, non-subnormal)");
```

Comment:
Why did you write this? An exponent can be zero (excepted 0^0, which is a specific case)

Reply:
You are absolutely correct, I misremembered that rust had strict requirements regarding pow, which it in fact does not. Please note however that rust defines 0^0 as 0.